pkg/devfile/adapters/kubernetes/component/adapter.go

```diff
@@ -693,40 +692,36 @@ func (a Adapter) deleteRemoteResources(objectsToRemove []unstructured.Unstructur
 		resources = append(resources, fmt.Sprintf("%s/%s", u.GetKind(), u.GetName()))
 	}
 
-	var err error
-	spinner := log.Spinnerf("Deleting resources not present in the Devfile: %s", strings.Join(resources, ", "))
```

Comment:
I assume we're sacrificing the spinner message to be able to use errgroup and make sure we don't run into race conditions, yes?

Reply:
We could have used `errgroup` with the spinner, as the spinner is started outside of the group and its status is updated after all goroutines have exited.
The problem is that `odo` will continuously try to delete such resources that it couldn't delete (and display the spinner over and over again), which adds too much clutter to the terminal, IMO.
Also, I found the final status of the spinner to be a bit misleading by displaying `✓`, because it ignored `PodMetrics` resources (in the example below), but it didn't delete them actually as it will try again to delete them :-).

<details>
<summary>Sample output with the spinner</summary>

```
$ odo dev                                                                                                                                                                                                           
  __                                                                                                                                                                                                                
 /  \__     Developing using the "my-node-app" Devfile                                                                                                                                                              
 \__/  \    Namespace: default                                                                                                                                                                                      
 /  \__/    odo version: v3.8.0                                                                                                                                                                                     
 \__/                                                                                                                                                                                                               
                                                                                                                                                                                                                    
↪ Running on the cluster in Dev mode                                                                                                                                                                                
 •  Waiting for Kubernetes resources  ...                                                                                                                                                                           
 ✓  Creating resource Pod/k8s-deploybydefault-true-and-not-referenced                                                                                                                                               
 ✓  Creating resource Pod/k8s-deploybydefault-false-and-not-referenced                                                                                                                                              
 ✓  Creating resource Pod/ocp-deploybydefault-true-and-not-referenced                                                                                                                                               
 ✓  Creating resource Pod/k8s-deploybydefault-not-set-and-not-referenced                                                                                                                                            
 ✓  Creating resource Pod/ocp-deploybydefault-false-and-not-referenced                                                                                                                                              
 ✓  Creating resource Pod/ocp-deploybydefault-not-set-and-not-referenced                                                                                                                                            
 ⚠  Pod is Pending                                                                                                                                                                                                  
 ✓  Pod is Running                                                                                                                                                                                                  
 ✓  Deleting resources not present in the Devfile: PodMetrics/k8s-deploybydefault-false-and-not-referenced, PodMetrics/k8s-deploybydefault-not-set-and-not-referenced, PodMetrics/k8s-deploybydefault-true-and-not-referenced, PodMetrics/ocp-deploybydefault-false-and-not-referenced, PodMetrics/ocp-deploybydefault-not-set-and-not-referenced, PodMetrics/ocp-deploybydefault-true-and-not-referenced [186ms]
 ✓  Syncing files into the container [101ms]
 ✓  Building your application in container (command: build) [3s]
 •  Executing the application (command: start-app)  ...
 -  Forwarding from 127.0.0.1:20001 -> 3000


↪ Dev mode
 Status:
 Watching for changes in the current directory /home/asoro/work/tmp/5694-implement-autobuild-and-deploybydefault

 Keyboard Commands:
[Ctrl+c] - Exit and delete resources from the cluster 
     [p] - Manually apply local changes to the application on the cluster
 ✓  Deleting resources not present in the Devfile: PodMetrics/k8s-deploybydefault-false-and-not-referenced, PodMetrics/k8s-deploybydefault-not-set-and-not-referenced, PodMetrics/k8s-deploybydefault-true-and-not-referenced, PodMetrics/ocp-deploybydefault-false-and-not-referenced, PodMetrics/ocp-deploybydefault-not-set-and-not-referenced, PodMetrics/ocp-deploybydefault-true-and-not-referenced [215ms]
 ✓  Deleting resources not present in the Devfile: PodMetrics/k8s-deploybydefault-false-and-not-referenced, PodMetrics/k8s-deploybydefault-not-set-and-not-referenced, PodMetrics/k8s-deploybydefault-true-and-not-referenced, PodMetrics/ocp-deploybydefault-false-and-not-referenced, PodMetrics/ocp-deploybydefault-not-set-and-not-referenced, PodMetrics/ocp-deploybydefault-true-and-not-referenced [839ms]
 ✓  Deleting resources not present in the Devfile: PodMetrics/k8s-deploybydefault-false-and-not-referenced, PodMetrics/k8s-deploybydefault-not-set-and-not-referenced, PodMetrics/k8s-deploybydefault-true-and-not-referenced, PodMetrics/ocp-deploybydefault-false-and-not-referenced, PodMetrics/ocp-deploybydefault-not-set-and-not-referenced, PodMetrics/ocp-deploybydefault-true-and-not-referenced [860ms]
 ✓  Deleting resources not present in the Devfile: PodMetrics/k8s-deploybydefault-false-and-not-referenced, PodMetrics/k8s-deploybydefault-not-set-and-not-referenced, PodMetrics/k8s-deploybydefault-true-and-not-r
eferenced, PodMetrics/ocp-deploybydefault-false-and-not-referenced, PodMetrics/ocp-deploybydefault-not-set-and-not-referenced, PodMetrics/ocp-deploybydefault-true-and-not-referenced [853ms]
 ✓  Deleting resources not present in the Devfile: PodMetrics/k8s-deploybydefault-false-and-not-referenced, PodMetrics/k8s-deploybydefault-not-set-and-not-referenced, PodMetrics/k8s-deploybydefault-true-and-not-r
eferenced, PodMetrics/ocp-deploybydefault-false-and-not-referenced, PodMetrics/ocp-deploybydefault-not-set-and-not-referenced, PodMetrics/ocp-deploybydefault-true-and-not-referenced [849ms]
 ✓  Deleting resources not present in the Devfile: PodMetrics/k8s-deploybydefault-false-and-not-referenced, PodMetrics/k8s-deploybydefault-not-set-and-not-referenced, PodMetrics/k8s-deploybydefault-true-and-not-r
eferenced, PodMetrics/ocp-deploybydefault-false-and-not-referenced, PodMetrics/ocp-deploybydefault-not-set-and-not-referenced, PodMetrics/ocp-deploybydefault-true-and-not-referenced [855ms]

...
```

</details>

This is why I thought it would make sense to `klog` this instead.